tests/test_connection.py

```diff
@@ -96,7 +96,7 @@ def test_reconnect_channel_peers(node_factory, executor):
     wait_for(lambda: not only_one(l1.rpc.listpeers(l2.info['id'])['peers'])['connected'])
 
     # Now should fail.
-    with pytest.raises(RpcError, match=r'Connection refused'):
+    with pytest.raises(RpcError, match=r'(Connection refused|Bad file descriptor)'):
         l1.rpc.connect(l2.info['id'], 'localhost', l2.port)
```

Comment:
Looks like this broke CI. Python regex issue, not sure this syntax is correct?

Also, we now run CI on MacOS and have not seen this? I cannot obviously see why it would get Bad file descriptor unless we have a bug?

Reply:
The Mac OS CI task doesn't run tests at the moment, exactly because there is a lot of breakage in the tests when we do. So this type of fix will eventually get us to running tests on Mac OS too.